edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/RegistryOnBehalfOfApiClient.cs

```diff
@@ -0,0 +1,167 @@
+            return response;
+        }
+
+        public async Task<RegistryApiHttpResult> ListModulesAsync(string actorEdgeDeviceId, ListModulesOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, this.listModulesOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
+            var response = await this.SendRequestAsync(requesteUri, HttpMethod.Post, requestData);
+            return response;
+        }
+
+        public async Task<RegistryApiHttpResult> DeleteModuleAsync(string actorEdgeDeviceId, DeleteModuleOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, this.deleteModuleOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
+            var response = await this.SendRequestAsync(requesteUri, HttpMethod.Post, requestData);
+            return response;
+        }
+
+        async Task<RegistryApiHttpResult> SendRequestAsync<T>(Uri uri, HttpMethod method, T payload, string ifMatchHeader = null)
+        {
+            string token = await this.tokenProvider.GetTokenAsync(Option.None<TimeSpan>());
+
+            using (HttpClient client = this.proxy
+                .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
+                .GetOrElse(() => new HttpClient()))
+            using (var request = new HttpRequestMessage(method, uri))
+            {
+                string jsonPayload = JsonConvert.SerializeObject(payload);
+                using (var stringContent = new StringContent(jsonPayload, Encoding.UTF8, "application/json"))
+                {
+                    request.Headers.Add(HttpRequestHeader.Authorization.ToString(), token);
+                    if (!string.IsNullOrEmpty(ifMatchHeader))
+                    {
+                        request.Headers.Add("if-match", ifMatchHeader);
+                    }
+
+                    request.Content = stringContent;
+
+                    using (var response = await client.SendAsync(request))
+                    {
+                        string jsonContent = string.Empty;
+                        if (response.Content != null)
+                        {
+                            jsonContent = await response.Content.ReadAsStringAsync();
+                        }
+
+                        return new RegistryApiHttpResult(response.StatusCode, jsonContent);
+                    }
+                }
+            }
+        }
+    }
+
+    public class RegistryApiHttpResult
+    {
+        public RegistryApiHttpResult(HttpStatusCode statusCode, string jsonContent)
+        {
+            this.StatusCode = statusCode;
+            this.JsonContent = jsonContent ?? string.Empty;
+        }
+
+        public HttpStatusCode StatusCode { get; private set; }
+
+        public string JsonContent { get; private set; }
+    }
+
+    public class CreateOrUpdateModuleOnBehalfOfData
```

Comment:
Do we need to define all these types? WHy not make them parameters on the call itself? Seems like its just a few parameters anyways

Reply:
This is the payload and need serialization/deserialization.  Keep as-is for now.